app/models/dps_export.rb

```diff
@@ -27,26 +27,36 @@ class DPSExport < ApplicationRecord
   after_create :set_filename
 
   belongs_to :campaign
+  has_and_belongs_to_many :vaccination_records
 
   def csv
     @csv ||=
       CSV.generate(headers: true, force_quotes: true) do |csv|
         csv << DPSExportRow::FIELDS.map(&:upcase)
 
-        unexported_vaccination_records.each { csv << DPSExportRow.new(_1).to_a }
+        vaccination_records.each { csv << DPSExportRow.new(_1).to_a }
       end
   end
 
+  def populate!
+    self.vaccination_records =
+      VaccinationRecord
+        .includes(:batch, :location, :patient, :session, :team, :vaccine)
```

Comment:
Is the `includes` here in the assignment so when we call `vaccination_records` again later in `export!` it's already fetched from the database?

I wonder if there's a way of making that clearer, or supporting the case where `populate!` hasn't been called first.

For example doing `DPSExport.find(...).csv` would have N+1 issues as it wouldn't have associated included.

Reply:
Would it make sense to set `vaccination_records` in the `initialize`?

Reply:
Have updated this and pushed. I'm really stumped why the test is failing though. It says it violates the uniqueness constraint, but I don't get why that's getting triggered.